jupyter-lib/shared-compiler/src/main/kotlin/org/jetbrains/kotlinx/jupyter/debug/DebugIntegration.kt

```diff
@@ -0,0 +1,21 @@
+package org.jetbrains.kotlinx.jupyter.debug
+
+import kotlin.concurrent.thread
+
+abstract class DebugIntegrationProvider(
+    protected val debugPort: Int?,
+) {
+    companion object {
+        const val DEBUG_THREAD_NAME = "NotebookDebugThread"
+    }
+
+    fun provideDebugPort(): Int? = debugPort
```

Comment:
It looks like debugPort doesn't need wrappers like this

Reply:
Or maybe we don't need to provide debug port here at all? It looks like the logic is boolean here: if we want or don't want to create a debug thread